apiserver/charms.go

```diff
@@ -401,11 +402,25 @@ func (h *charmsHandler) repackageAndUploadCharm(st *state.State, archive *charm.
 	if err := archive.ExpandTo(extractPath); err != nil {
 		return errors.Annotate(err, "cannot extract uploaded charm")
 	}
+
 	charmDir, err := charm.ReadCharmDir(extractPath)
 	if err != nil {
 		return errors.Annotate(err, "cannot read extracted charm")
 	}
 
+	// try to get the version details here.
+	// we do not want to expose it to charm repo.
+	// read just the first line of the file.
+	var version string
+	versionPath := filepath.Join(extractPath, "version")
+	if file, err := os.Open(versionPath); err == nil {
```

Comment:
We need to check the error. If it is due to file not found, that's ok, we can skip trying to read the file. But if there's another sort of error, we need to report it.